### ISSUE TITLE
The confusion about code

### ISSUE THREAD

**Issue by fuyuyuputao** [EnyanDai/GANF#8]
Hello, thank you for making the code available, it's a very nice work.  When I run train_water.py, I have a confusion on line 66 of code in GANF.py. In the paper, the log(p(x)) in equation (10) is sum() but in the code it becomes mean(). I can't understand why this change was introduced. I have tried to change mean() to sum(), I find the result about the best roc_test is 0.7875 that satisfies 79.6±0.9 in Table 1. But if I don't change it, the result will be better that the roc_test will be 0.79 or 0.80. I can't understand why this change was introduced. Thank you!

![6@ZN`GPZZ6@ZYE}RF$%9NYT](https://user-images.githubusercontent.com/79823933/186419301-92fac08e-ae85-49a2-a3bd-a9db2cd13f45.png)

I think log_prob = log_prob.sum(dim=1) are more reasonable.
```
class GANF(nn.Module):

    def __init__ (self, n_blocks, input_size, hidden_size, n_hidden ,dropout = 0.1, model="MAF", batch_norm=True):
        super(GANF, self).__init__()

        self.rnn = nn.LSTM(input_size=input_size,hidden_size=hidden_size,batch_first=True, dropout=dropout)
        self.gcn = GNN(input_size=hidden_size, hidden_size=hidden_size)
        if model=="MAF":
            self.nf = MAF(n_blocks, input_size, hidden_size, n_hidden, cond_label_size=hidden_size, batch_norm=batch_norm,activation='tanh')
        else:
            self.nf = RealNVP(n_blocks, input_size, hidden_size, n_hidden, cond_label_size=hidden_size, batch_norm=batch_norm)

    def forward(self, x, A):

        return self.test(x, A).mean()

    def test(self, x, A):
        # x: N X K X L X D 
        full_shape = x.shape

        # reshape: N*K, L, D
        x = x.reshape((x.shape[0]*x.shape[1], x.shape[2], x.shape[3]))
        h,_ = self.rnn(x)

        # resahpe: N, K, L, H
        h = h.reshape((full_shape[0], full_shape[1], h.shape[1], h.shape[2]))


        h = self.gcn(h, A)

        # reshappe N*K*L,H
        h = h.reshape((-1,h.shape[3]))
        x = x.reshape((-1,full_shape[3]))

        log_prob = self.nf.log_prob(x,h).reshape([full_shape[0],-1])#*full_shape[1]*full_shape[2]
        log_prob = log_prob.mean(dim=1)

        return log_prob
```

**Comment by EnyanDai** [comment]
Hi,

Thanks very much for your interests to our work. There is no much difference between sum and mean here as they all try to maximize the probability. The only difference is that ratio of the the log_prob and the DAG constraint is slightly different, resulting slightly different results. Hope this address your concern.

**Comment by fuyuyuputao** [comment]
Thank you very much!
